app/controllers/proposals_controller.rb

```diff
@@ -148,4 +149,28 @@ def require_waitlisted_or_accepted_state
       redirect_to event_url(@event.slug)
     end
   end
+
+  def require_complete_profile
+    return if current_user.complete? && current_user.unconfirmed_email.blank?
```

Comment:
When I see a line like this, I ask myself if this controller knows too much about the model.  In this case, I think it does because the method's intention is clear.  Check for a complete profile.  Moving the unconfirmed email into User#complete? check would be ideal.